src/lib/phdi-building-blocks/phdi_building_blocks/standardize.py

```diff
@@ -1,93 +1,172 @@
+    family_name_altered = False
+    given_name_altered = False
 
-def country_extractor(
-    resource: dict, code_type: Literal["alpha_2", "alpha_3", "numeric"] = "alpha_2"
-) -> List[str]:
+    for name in patient.get("name", []):
+
+        # Handle family names
+        if "family" in name:
+            transformed_name = transform_func(name.get("family", ""))
+            family_name_altered = family_name_altered or (
+                transformed_name != name.get("family", "")
+            )
+            name["family"] = transformed_name
+
+        # Given names are stored in a list, as there could be multiple,
+        # so process them all and take the overall diff for metrics
+        if "given" in name:
+            transformed_names = [transform_func(g) for g in name.get("given", [])]
+            given_name_altered = any(
+                [
+                    name.get("given", [])[i] != transformed_names[i]
+                    for i in range(len(name.get("given", [])))
+                ]
+            )
+            name["given"] = transformed_names
+
+    if add_metrics_extensions:
+        add_name_change_metrics_values(patient, given_name_altered, family_name_altered)
+
+
+def add_name_change_metrics_values(
+    patient: dict, given_name_was_changed: bool, family_name_was_changed: bool
+) -> None:
+    """
+    Add information about whether a patient's given or family name was
+    changed through the use of our standardization function directly
+    to the patient resource as an extension. We define a pseudo-FHIR standard
+    URL for this purpose.
+    :param dict patient: The Patient to add data for
+    :param bool given_name_was_changed: Did the transforms standardize
+    the given name string to something different than the raw form
+    :param bool family_name_was_Changed: Did the transforms standardize
+    the family name string to something different than the raw form
+    :return: None
     """
-    Given a FHIR resource yield a generator containing all of the counries found in the
-    resource in a standard form sepcified by code_type.
+    # Add in the tracking array for extra properties, if not there
+    if "extension" not in patient:
+        patient["extension"] = []
+
+    patient["extension"].append(
+        {
+            "url": "http://usds.gov/fhir/phdi/StructureDefinition/given-name-was-standardized",  # noqa
+            "valueBoolean": given_name_was_changed,
+        }
+    )
+    patient["extension"].append(
+        {
+            "url": "http://usds.gov/fhir/phdi/StructureDefinition/family-name-was-standardized",  # noqa
+            "valueBoolean": family_name_was_changed,
+        }
+    )
 
-    We currently only handle extracting countries from patient resources, however as
-    need arises is the future funcionality can be expanded.
 
-    :param dict resource: A FHIR resource respresented as a dictionary.
+def country_extractor(
```

Comment:
I think we should leave this function generalized for all resources (I can see a scenario where a user would want to standardize the phone number for a Physician), pass in the full bundle, allow the end user to specify the resource they want to target, and use the `find_<x>_resources()` function that's been defined to narrow down to just that. 

We're effectively taking a step backward with this change by making it less generalized than it already was.

Reply:
I've now decided that this is slightly out of scope for this PR. This isn't related to human-centered design, per se, and is more related to generalization and scaling. Going to mark as resolved.

Reply:
After looking at the code and how we're using this function, I agree strongly with Dan. The function was written for the most general usage our particular case requires--we can already extract countries from any object. Bryan, where I think we run into a problem with the potential implementation you want to see in the future (i.e. passing in a bundle and using `find_resource_of_type`) is that that will return a list of _all_ resources of the given type in the bundle, but we want to build up a list of countries for a single _individual_ resource. I.e. we don't want the returned list to have all countries that occur across all addresses in the bundle, but rather just the countries that occur for one specific resource. Consider a bundle with two patients, X and Y, each with two addresses: X has country USA and country UK, and Y has country USA and country CA. The returned list would have three elements, which would then cause problems on line 210 when we call `phone_country_standardiation` for each individual patient and there are only two relevant elements in each patient list. I'm going to leave the existing implementation as it is with only a revised name for clarity, because it seems the most general use that balances between both of your desires, but we can open a TODO or fast follows if we really want to change it.

Reply:
That makes sense to me. The solution then is to revert this back to how Dan wrote it originally, right?

Reply:
This function has been reverted, so all is well.